staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -165,6 +171,10 @@ func (s *ServerRunOptions) AddUniversalFlags(fs *pflag.FlagSet) {
 		"will be used. If --bind-address is unspecified, the host's default interface will "+
 		"be used.")
 
+	fs.IntVar(&s.AdvertisePort, "advertise-port", s.AdvertisePort, ""+
+		"The port on which to advertise the apiserver to members of the cluster. If blank, "+
+		"the --secure-port will be used.")
```

Comment:
So, secure-port is what we will attempt to listen on, but advertise-port is what we will put in the endpoints record?